NPE encountered during Dubbo shutdown

版本： Dubbo 3.0.4 + Nacos 2.0.3
情景：本地没启动 Nacos Server，所以 Dubbo 启动后报错进入销毁流程，销毁过程中 Nacos 报了以下NPE。

```java
[13/12/21 11:54:31:031 CST] DubboShutdownHook  WARN support.RegistryManager:  [DUBBO] java.lang.NullPointerException, dubbo version: 3.0.4, current host: 192.168.0.105
java.lang.RuntimeException: java.lang.NullPointerException
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:48)
	at org.apache.dubbo.registry.client.ServiceDiscoveryRegistry.destroy(ServiceDiscoveryRegistry.java:277)
	at org.apache.dubbo.registry.support.RegistryManager.destroyAll(RegistryManager.java:105)
	at org.apache.dubbo.config.deploy.DefaultApplicationDeployer.destroyRegistries(DefaultApplicationDeployer.java:1034)
	at org.apache.dubbo.config.deploy.DefaultApplicationDeployer.postDestroy(DefaultApplicationDeployer.java:860)
	at org.apache.dubbo.rpc.model.ApplicationModel.onDestroy(ApplicationModel.java:262)
	at org.apache.dubbo.rpc.model.ScopeModel.destroy(ScopeModel.java:105)
	at org.apache.dubbo.config.DubboShutdownHook.doDestroy(DubboShutdownHook.java:67)
	at org.apache.dubbo.config.DubboShutdownHook.run(DubboShutdownHook.java:62)
Caused by: java.lang.NullPointerException
	at com.alibaba.nacos.common.remote.client.RpcClient.shutdown(RpcClient.java:465)
	at com.alibaba.nacos.common.remote.client.grpc.GrpcClient.shutdown(GrpcClient.java:81)
	at com.alibaba.nacos.client.naming.remote.gprc.NamingGrpcClientProxy.shutdown(NamingGrpcClientProxy.java:286)
	at com.alibaba.nacos.client.naming.remote.NamingClientProxyDelegate.shutdown(NamingClientProxyDelegate.java:180)
	at com.alibaba.nacos.client.naming.NacosNamingService.shutDown(NacosNamingService.java:456)
	at org.apache.dubbo.registry.nacos.NacosNamingServiceWrapper.shutdown(NacosNamingServiceWrapper.java:84)
	at org.apache.dubbo.registry.nacos.NacosServiceDiscovery.doDestroy(NacosServiceDiscovery.java:67)
	at org.apache.dubbo.registry.client.AbstractServiceDiscovery.destroy(AbstractServiceDiscovery.java:81)
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:46)
	... 8 more
```

--- a/common/src/main/java/com/alibaba/nacos/common/remote/client/RpcClient.java
+++ b/common/src/main/java/com/alibaba/nacos/common/remote/client/RpcClient.java
@@ -454,7 +454,9 @@ public abstract class RpcClient implements Closeable {
         LOGGER.info("Shutdown rpc client, set status to shutdown");
         rpcClientStatus.set(RpcClientStatus.SHUTDOWN);
         LOGGER.info("Shutdown client event executor " + clientEventExecutor);
-        clientEventExecutor.shutdownNow();
+        if (clientEventExecutor != null) {
+            clientEventExecutor.shutdownNow();
+        }
         closeConnection(currentConnection);
     }
     

--- a/common/src/test/java/com/alibaba/nacos/common/remote/client/RpcClientTest.java
+++ b/common/src/test/java/com/alibaba/nacos/common/remote/client/RpcClientTest.java
@@ -21,6 +21,7 @@ import com.alibaba.nacos.api.remote.RequestCallBack;
 import com.alibaba.nacos.api.remote.request.Request;
 import com.alibaba.nacos.api.remote.response.ErrorResponse;
 import com.alibaba.nacos.common.remote.ConnectionType;
+import com.alibaba.nacos.common.remote.client.grpc.GrpcClient;
 import com.alibaba.nacos.common.remote.client.grpc.GrpcConnection;
 import org.junit.After;
 import org.junit.Assert;
@@ -388,5 +389,27 @@ public class RpcClientTest {
         verify(rpcClient).switchServerAsyncOnRequestFail();
         Assert.assertNotNull(exception);
         Assert.assertEquals(RpcClientStatus.UNHEALTHY, rpcClient.rpcClientStatus.get());
+    }
+    
+    @Test
+    public void testRpcClientShutdownWhenClientDidntStart() throws NacosException {
+        RpcClient rpcClient = new RpcClient("test-client") {
+            @Override
+            public ConnectionType getConnectionType() {
+                return null;
+            }
+    
+            @Override
+            public int rpcPortOffset() {
+                return 0;
+            }
+    
+            @Override
+            public Connection connectToServer(ServerInfo serverInfo) throws Exception {
+                return null;
+            }
+        };
+        
+        rpcClient.shutdown();
     }
 }

--- a/common/src/test/java/com/alibaba/nacos/common/remote/client/RpcClientTest.java
+++ b/common/src/test/java/com/alibaba/nacos/common/remote/client/RpcClientTest.java
@@ -21,7 +21,6 @@ import com.alibaba.nacos.api.remote.RequestCallBack;
 import com.alibaba.nacos.api.remote.request.Request;
 import com.alibaba.nacos.api.remote.response.ErrorResponse;
 import com.alibaba.nacos.common.remote.ConnectionType;
-import com.alibaba.nacos.common.remote.client.grpc.GrpcClient;
 import com.alibaba.nacos.common.remote.client.grpc.GrpcConnection;
 import org.junit.After;
 import org.junit.Assert;

